File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.js
/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function (s, spaces) {
    // let ss=s.split("");

    // for(let i =0;i<s.length;i++){
    //     if(spaces.includes(i)){
    //         ss.splice(i,1," ");
    //     }
    // }
    // return ss.join("");

    let ret = [];
    let jj = 0;
    for (let i = 0; i < s.length; i++) {
        if (i == spaces[jj]) {
            ret.push(" ");
            j = spaces[jj++];
        }

        ret.push(s[i]);
    }
    return ret.join("");
};

File: 2109-adding-spaces-to-a-string/2109-adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder ret = new StringBuilder();
        int jj = 0;
        for (int i = 0; i < s.length(); i++) {
            if (jj < spaces.length) {
                if (i == spaces[jj]) {
                    ret.append(" ");
                    jj++;
                }
            }
            ret.append(s.charAt(i));
        }

        return ret.toString();
    }
}